src/match/querymatch.c

```diff
@@ -503,12 +503,18 @@ double gt_querymatch_error_rate(GtUword distance,GtUword alignedlen)
 }
 
 bool gt_querymatch_overlap(const GtQuerymatch *querymatch,
-                           GtUword start_relative)
+                           GtUword nextseed_db_end_relative,
+                           GtUword nextseed_query_end_relative,
+                           bool use_db_pos)
 {
   gt_assert(querymatch != NULL);
-  return querymatch->querystart + querymatch->querylen >= start_relative
-           ? true /* overlap with querymatch */
-           : false; /* next seeds starts after curren extension */
+  return querymatch->querystart + querymatch->querylen >
+         nextseed_query_end_relative && (!use_db_pos ||
+         (querymatch->dbstart_relative + querymatch->dblen >
+          nextseed_db_end_relative && nextseed_db_end_relative >=
+          querymatch->dbstart_relative + querymatch->seedlen))
+           ? true   /* overlap with querymatch */
+           : false; /* next seed ends after current extension */
```

Comment:
That code is quite convoluted... I think in future cases it might help not trying to do everything in one expression. For the compiler it most likely doesn't matter anyway, but if it was my code I know my future self would thank me later ;)